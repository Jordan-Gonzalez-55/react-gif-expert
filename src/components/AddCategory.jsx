import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    //Aqui es otro ejemplo pero cambiandolo desde el componente hijo
    //setCategories((categories) => [inputValue, ...categories]);

    //Aqui es para cambiarlo desde el componente padre
    onNewCategory(inputValue.trim());
    setInputValue("");
  };
  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
