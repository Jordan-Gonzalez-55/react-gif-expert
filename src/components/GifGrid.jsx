import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  /*const [images, setImages] = useState([]);

  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };

  useEffect(() => {
    getImages();
  }, []);
*/
  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GifItem
            key={image.id}
            //Aqui le estamos pasando todas las propiedades de putaso y en el otro componente las destructuramos
            {...image}
          />
        ))}
      </div>
    </>
  );
};
