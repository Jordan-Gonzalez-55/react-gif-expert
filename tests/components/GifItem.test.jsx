import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Purebas en <GiftItem />", () => {
  const title = "Saitama";
  const url = "https://media.giphy.com/media/l4qR5hjZt8yFv3";
  test("debe hacer match con el snaptshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
