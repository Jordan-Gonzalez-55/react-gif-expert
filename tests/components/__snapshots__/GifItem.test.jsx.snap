// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purebas en <GiftItem /> debe hacer match con el snaptshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media.giphy.com/media/l4qR5hjZt8yFv3"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
